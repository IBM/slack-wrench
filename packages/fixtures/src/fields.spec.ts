import { actions, fields } from './index';

describe('Defaults', () => {
  afterEach(() => {
    fields.reset();
  });

  it('has all the fields', () => {
    expect.assertions(1);
    expect(fields).toMatchSnapshot();
  });

  it('allows global overrides', () => {
    expect.assertions(2);
    const overrideToken = 'a-very-different-token';
    const originalToken = fields.token;

    expect(actions.blockButtonAction()).toEqual(
      expect.objectContaining({
        token: originalToken,
      }),
    );

    fields.token = overrideToken;

    expect(actions.blockButtonAction()).toEqual(
      expect.objectContaining({
        token: overrideToken,
      }),
    );
  });

  it('Allows resetting overrides', () => {
    expect.assertions(1);
    const originalTs = fields.ts;

    fields.ts = "this isn't a real ts";

    fields.reset();

    expect(fields.ts).toEqual(originalTs);
  });
});
