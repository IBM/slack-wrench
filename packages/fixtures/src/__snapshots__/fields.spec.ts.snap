// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Defaults has all the fields 1`] = `
Fields {
  "action_id": "ACTION_ID",
  "action_ts": "0000000000.000000",
  "api_app_id": "API_APP_ID",
  "authed_users": Array [
    "UUSERID",
  ],
  "block_id": "BLOCK_ID",
  "callback_id": "CALLBACK_ID",
  "channel": Object {
    "id": "CCHANNELID",
    "name": "channel",
  },
  "event_id": "EVENT_ID",
  "event_time": 1234567890,
  "response_url": "https://fake.slack/response_url",
  "team": Object {
    "domain": "team-domain",
    "id": "TTEAMID",
  },
  "text": "TEXT",
  "token": "TOKEN",
  "trigger_id": "TRIGGER_ID",
  "ts": "0000000000.000000",
  "user": Object {
    "id": "UUSERID",
    "name": "USER",
  },
  "value": "VALUE",
}
`;
